### PR TITLE
Adds ability to specify CPU credit type for NAT instances

### PR DESCRIPTION
**Pull request by bradj**
## what
* Module allows user to specify either `standard`, `unilimited`, or the default credit specification for NAT instances
* Updates to the credit specification will update the instance in place

## why
* Provides more predictable bills
* No longer need to worry about how much bursting could cost

## references
* https://aws.amazon.com/premiumsupport/knowledge-center/ec2-calculate-burst-credit-billing/
* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/burstable-performance-instances-unlimited-mode.html

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -82,6 +82,12 @@ variable "nat_instance_type" {
   default     = "t3.micro"
 }
 
+variable "nat_instance_cpu_credits_override" {
+  type        = string
+  description = "NAT Instance credit option for CPU usage. Valid values include standard or unlimited. T3 instances are launched as unlimited by default. T2 instances are launched as standard by default."
```
Comment (nitrocode) [inline variables.tf:87]: can you add validation here

Comment (bradj) [reply variables.tf:87]: @nitrocode done

**Review by Gowiem — APPROVED** [review]

File: variables.tf [hunk 2]
```diff
@@ -82,6 +82,17 @@ variable "nat_instance_type" {
...
+  default     = ""
+
+  validation {
+    condition     = contains(["standard", "unlimited"], var.nat_instance_cpu_credits_override)
```
Comment (Gowiem) [inline variables.tf:91]: @bradj This validation is failing because the default is neither of these. Mind accounting for that?

**Review by Gowiem — APPROVED** [review]

**Pull request merged**